### PR TITLE
Race entry not working

### PR DESCRIPTION
**Pull request by moveson**
Fixed race entry form by changing `<% %>` delimiters to `<%= %>` in `form_for` and `fields_for` tags. Added many database-level constraints to backstop Rails validations. Added #paid attribute to the RaceEntry model and fixed related methods.

### CODE REVIEW

File: app/models/race_entry.rb [hunk 1]
```diff
@@ -3,21 +3,18 @@
...
+  validates_uniqueness_of :racer_id, scope: :race_edition_id,
+                          message: 'may be added to a race_edition only once'
 
   def elapsed_time
-    if time? then
-      ChronicDuration.output(time)
-    else
-      'NA'
-    end
+    time && ChronicDuration.output(time) || 'NA'
```
Comment (moveson) [inline app/models/race_entry.rb]: In Ruby everything evaluates to truthy or falsey, allowing shortcuts like this. if `time` is nil, then we pass through the `||` operator and return "NA".

File: app/models/race_entry.rb [hunk 2]
```diff
@@ -3,21 +3,18 @@
...
   end
-  
-  def set_elapsed_time(time_string)
-    time = ChronicDuration.parse(time_string)
+
+  def elapsed_time=(time_string)
+    time_string ||= ''
+    self.time = ChronicDuration.parse(time_string)
   end
```
Comment (moveson) [inline app/models/race_entry.rb]: By renaming this method in this way, we create a "virtual attribute" on the RaceEntry model. Now we can (for example) pass an `elapsed_time` parameter from a form and the `time` attribute will be automatically set.

File: app/controllers/race_editions_controller.rb [hunk 3]
```diff
@@ -1,50 +1,66 @@
...
+
+
+  private
+
+  def obj_params
+    params.require(:race_edition)
+        .permit(:date, racers_attributes: [:id, :first_name, :last_name, :email, :gender, :birth_date, :city, :state])
+  end
+
+  def set_race_edition
```
Comment (moveson) [inline app/controllers/race_editions_controller.rb]: This method, together with the before_action on line 2, allows us to DRY out the controller

File: app/models/race_edition.rb [hunk 4]
```diff
@@ -1,12 +1,15 @@
...
-   def name
-      race.name + ':' + date.to_s
-   end
-   
- end
\ No newline at end of file
+  belongs_to :race
+  has_many :race_entries
+  has_many :racers, through: :race_entries
+
```
Comment (moveson) [inline app/models/race_edition.rb]: This `has_many: through` relationship is very convenient as it allows us to deal with racers by calling `race_edition.racers` and letting ActiveRecord create the multi-join query for us.

**Comment by billwright** [comment]
Great stuff, Mark! Thanks a ton for the help and explanations. All looks great, except for the migrations. Before we have any actual data, I think that isn't the right approach and makes things overly complex. Of course, once you go into production, that is the way to go. I assume that all those migrations get built into the schema.rb file? How does that happen again? Some rake task? Then I can just copy what's in there to my model schema files, right?

**Pull request merged**